src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -111,4 +120,56 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
             StartingDirectory().Text(folder.Path());
         }
     }
+
+    bool Profiles::_IsCustomFontWeight(const Windows::UI::Text::FontWeight& weight)
+    {
+        if (weight == FontWeights::Thin() ||
+            weight == FontWeights::ExtraLight() ||
+            weight == FontWeights::Light() ||
+            weight == FontWeights::SemiLight() ||
+            weight == FontWeights::Normal() ||
+            weight == FontWeights::Medium() ||
+            weight == FontWeights::SemiBold() ||
+            weight == FontWeights::Bold() ||
+            weight == FontWeights::ExtraBold() ||
+            weight == FontWeights::Black() ||
+            weight == FontWeights::ExtraBlack())
+        {
+            return false;
+        }
+        else
+        {
+            return true;
+        }
+    }
+
+    IInspectable Profiles::CurrentFontWeight()
+    {
+        // if no value was found, we have a custom value, and we show the slider.
+        const auto maybeEnumEntry{ _FontWeightMap.TryLookup(_State.Profile().FontWeight().Weight) };
+        CustomFontWeightControl().Visibility(maybeEnumEntry ? Visibility::Collapsed : Visibility::Visible);
+        return maybeEnumEntry ? maybeEnumEntry : _CustomFontWeight;
```

Comment:
really don't love the fact that _the getter for the current font weight_ also impacts the UI's visibility. That's not what a getter is for.

Reply:
It makes sense though, no? Because the visibility of the slider is dependent on the value. So we change it when we get the value? Curious what other approach you suggest though